### PR TITLE
Move Concept Casting Utilities To Interfaces 

### PR DESCRIPTION
**Pull request by fppt**
This is needed as part of the resource edge change

### CODE REVIEW

File: grakn-graph/src/main/java/ai/grakn/graph/internal/AbstractGraknGraph.java [hunk 1]
```diff
@@ -470,12 +470,12 @@ public RelationType putRelationType(String label) {
     @Override
     public RelationType putRelationType(Label label) {
         return putOntologyElement(label, Schema.BaseType.RELATION_TYPE,
-                v -> factory().buildRelationType(v, getMetaRelationType(), Boolean.FALSE)).asRelationType();
```
Comment (aelred) [inline grakn-graph/src/main/java/ai/grakn/graph/internal/AbstractGraknGraph.java]: So good!!!!

File: grakn-graph/src/main/java/ai/grakn/graph/internal/TypeImpl.java [hunk 2]
```diff
@@ -307,12 +307,8 @@ public T deleteScope(Thing scope) {
     }
 
     void trackSuperChange(){
-        instances().forEach(concept -> {
-            if (concept.isInstance()) {
-                ((ThingImpl<?, ?>) concept).castingsInstance().forEach(
-                        rolePlayer -> vertex().graph().txCache().trackForValidation(rolePlayer));
-            }
-        });
```
Comment (aelred) [inline grakn-graph/src/main/java/ai/grakn/graph/internal/TypeImpl.java]: I'm so happy

**Comment by fppt** [comment]
!rtg

**Comment by fppt** [comment]
!rtg

**Pull request merged**